File: app/src/main/java/com99/example/gal/flexboxlayout/DivLikeAdapter.java
package com99.example.gal.flexboxlayout;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

class DivLikeAdapter extends RecyclerView.Adapter<DivLikeAdapter.DivItemViewHolder> {
    private ArrayList<String> mItems;

    public DivLikeAdapter(ArrayList<String> items) {
        mItems = items;
    }

    @NonNull
    @Override
    public DivLikeAdapter.DivItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item, parent, false);
        return new DivItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DivItemViewHolder holder, int position) {
        holder.onBind(position);
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }


    public class DivItemViewHolder extends RecyclerView.ViewHolder {
        private TextView text;

        private DivItemViewHolder(@NonNull View itemView) {
            super(itemView);
            text = itemView.findViewById(R.id.text);
        }

        private void onBind(int position) {
            text.setText(mItems.get(position));
        }
    }
}

File: app/src/main/java/com99/example/gal/flexboxlayout/DivLikeAdapter.kt
package com99.example.gal.flexboxlayout

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import java.util.ArrayList
import androidx.recyclerview.widget.RecyclerView

class DivLikeAdapter(private val mItems: ArrayList<String>) : RecyclerView.Adapter<DivLikeAdapter.DivItemViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DivLikeAdapter.DivItemViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.item, parent, false)
        return DivItemViewHolder(view)
    }

    override fun onBindViewHolder(holder: DivItemViewHolder, position: Int) {
        holder.onBind(position)
    }

    override fun getItemCount(): Int {
        return mItems.size
    }


    inner class DivItemViewHolder constructor(itemView: View) : RecyclerView.ViewHolder
    (itemView) {
        private val text: TextView = itemView.findViewById(R.id.text)

        fun onBind(position: Int) {
            text.text = mItems[position]
        }
    }
}